docs/docs/unit-testing.md

```diff
@@ -0,0 +1,302 @@
+    "@babel/plugin-proposal-class-properties",
+  ],
+}
+
+module.exports = require("babel-jest").createTransformer(babelOptions)
+```
+
+Back to the Jest config, you can see the next option is `testRegex`. This is the
+pattern telling Jest which files contain tests. The pattern above matches any
+`.js` file inside a `__tests__` directory, or any file elsewhere with the
+extension `.test.js` or `.spec.js`. You are telling Jest to ignore any tests in
+the `node_modules` or `.cache` directories.
+
+The `moduleNameMapper` section works a bit like webpack rules, and tells Jest
+how to handle imports. You are mainly concerned here with mocking static file
+imports, which Jest can't handle. A mock is a dummy module that is used instead
+of the real module inside tests. It is good when you have something that you
+can't or don't want to test. You can mock anything, and here you are mocking
+assets rather than code. For stylesheets you need to use the package
+`identity-obj-proxy`. For all other assets you need to use a manual mock called
+`fileMock.js`. You need to create this yourself. The convention is to create a
+directory called `__mocks__` in the root directory for this. Note the pair of
+double underscores in the name.
+
+```js
+// __mocks__/fileMock.js
+module.exports = "test-file-stub"
+```
+
+The next config setting is `transformIgnorePatterns`. This is very important,
+and is different from what you'll find in other Jest guides. The reason that you
+need this is because Gastby includes un-transpiled ES6 code. By default Jest
+doesn't try to transform code inside `node_modules`, so you will get an error
+like this:
+
+```
+/my-blog/node_modules/gatsby/cache-dir/gatsby-browser-entry.js:1
+({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import React from "react"
+                                                                                            ^^^^^^
+SyntaxError: Unexpected token import
+```
+
+This is because `gatsby-browser-entry.js` isn't being transpiled before running
+in Jest. You can fix this by changing the default `transformIgnorePatterns` to
+exclude the `gatsby` module.
+
+The `globals` section sets `__PATH_PREFIX__`, which is usually set by Gatsby,
+and which some components need.
+
+There's one more global that you need to set, but as it's a function you can't
+set it here in the JSON. The `setupFiles` array lets you list files that will be
+included before all tests are run, so it's perfect for this.
+
+```js
+// loadershim.js
+
+global.___loader = {
+  enqueue: jest.fn(),
+}
+```
+
+Finally it's a good idea to mock the gatsby module itself. This may not be
+needed at first, but will make things a lot easier if you want to test
+components that use `Link` or GraphQL.
+
+```js
+// __mocks__/gatsby.js
+"use strict"
+import React from "react"
+const gatsby = jest.genMockFromModule("gatsby")
+gatsby.Link = ({ to, ...props }) => <a href={to} {...props} />
```

Comment:
I encountered this issue because my NavItem component uses `activeClassName` on `Link`:

```
Warning: React does not recognize the `activeClassName` prop on a DOM element. If you intentionally want i
t to appear in the DOM as a custom attribute, spell it as lowercase `activeclassname` instead. If you accident
ally passed it from a parent component, remove it from the DOM element.
```

So we should write:

```JS
gatsby.Link = ({ to, activeClassName, activeStyle, ...props }) => (
  <a href={to} className={activeClassName} style={activeStyle} {...props} />
);
```